scss/components/button.scss

```diff
@@ -553,9 +553,12 @@ a.#{$block} {
     //DEPRECATE --secondary
     &--secondary {
       --fd-button-border-color: transparent;
+
       @include fd-hover() {
         --fd-button-background-color: var(--fd-color-neutral-1);
         --fd-button-color: var(--fd-color-action-1);
+        @include fd-var-color("background-color", map-get($fd-colors-neutral, 1),--fd-button-background-color);
+        @include fd-var-color("color", map-get($fd-colors-action, 1), --fd-button-color);
```

Comment:
I am trying to figure out _what_ this is changing/fixing.  I am not seeing a difference between `master` and this branch.

Also, the secondary button is marked as deprecated so is this change even necessary?

Reply:
w/o code at line 560 and 561, when hovered on the secondary button, shown as below screenshots:
self-contained style (left side of comparison playground): blue color
![Screen Shot 2019-07-17 at 9 33 06 AM](https://user-images.githubusercontent.com/41170611/61379988-93282600-a876-11e9-9908-ea6d732d8180.png)


left side of comparison playground: light grey color
![Screen Shot 2019-07-17 at 9 33 16 AM](https://user-images.githubusercontent.com/41170611/61379992-96231680-a876-11e9-9879-30c87c2e578b.png)

Reply:
@saad-mo Wanted to get your take on this.  The playground page shows "Secondary Button" as being deprecated.  Is this true?  Also, is there a difference between "Secondary" and "Default Light"?  It kind of looks like "Default Light" is meant to replace "Secondary".

Reply:
> @saad-mo Wanted to get your take on this. The playground page shows "Secondary Button" as being deprecated. Is this true? Also, is there a difference between "Secondary" and "Default Light"? It kind of looks like "Default Light" is meant to replace "Secondary".

`fd-button--secondary` is indeed deprecated. Fiori 3 guidelines called for a new naming convention which is `emphasized` (blue filled), `regular` (outline also same as secondary), and `light` (no borders). Along with semantic flavors of these three (standard, positive, medium, negative).

We left the deprecated version in the code to maintain backward compatibility and make non-breaking changes when introducing the new naming convention. I don't think we need to maintain this in fd-styles anymore since we are free to break things :)